op-conductor/conductor/config.go

```diff
@@ -0,0 +1,96 @@
+package conductor
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum-optimism/optimism/op-conductor/flags"
+	opnode "github.com/ethereum-optimism/optimism/op-node"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	oppprof "github.com/ethereum-optimism/optimism/op-service/pprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+)
+
+type Config struct {
+	// ConsensusAddr is the address to listen for consensus connections.
+	ConsensusAddr string
+
+	// ConsensusPort is the port to listen for consensus connections.
+	ConsensusPort int
+
+	// RaftServerID is the unique ID for this server used by raft consensus.
+	RaftServerID string
+
+	// RaftStorageDIR is the directory to store raft data.
+	RaftStorageDIR string
+
+	// RaftBootstrap is true if this node should bootstrap a new raft cluster.
+	RaftBootstrap bool
+
+	// NodeRPC is the HTTP provider URL for op-node.
+	NodeRPC string
+
+	RollupCfg rollup.Config
+
+	LogConfig     oplog.CLIConfig
+	MetricsConfig opmetrics.CLIConfig
+	PprofConfig   oppprof.CLIConfig
+	RPC           oprpc.CLIConfig
+}
+
+// Check validates the CLIConfig.
+func (c *Config) Check() error {
+	if c.ConsensusAddr == "" {
+		return fmt.Errorf("missing consensus address")
+	}
+	if c.ConsensusPort == 0 {
+		return fmt.Errorf("missing consensus port")
+	}
+	if c.RaftServerID == "" {
+		return fmt.Errorf("missing raft server ID")
+	}
+	if c.RaftStorageDIR == "" {
+		return fmt.Errorf("missing raft storage directory")
+	}
+	if c.NodeRPC == "" {
+		return fmt.Errorf("missing node RPC")
+	}
+	if err := c.RollupCfg.Check(); err != nil {
+		return err
```

Comment:
Why not wrap the errors below so it's easier to track down where they came from?

Reply:
Will do in the future, but the pattern I picked up from the rest of the codebase seems to not use wrapped errors.